### ISSUE TITLE
Hello o/ !

### ISSUE THREAD

**Issue by Sniperbg** [skullernet/opentdm#3]
Hello :) ! Can you check spawn start for problems.  Many times its very buggy, start one spawn 2-3 times next by next, or use only 2-3 spawns most of game. Its strange why it maybe need to be random, but don,t look at that. An its ridiculous when option avoid closest is on :D . Its not that obviously in 1v1 or 2v2  not everytime, but in 3v3 or 4v4.

**Comment by skullernet** [comment]
Quite a bit of effort has already been spent on fixing spawn point selection bugs in OpenTDM and I'm not aware of any existing bugs.

I've tested the `SelectRandomDeathmatchSpawnPoint` algorithm with simulated 1 billion spawns on a map with 10 spawn points and here are the numbers of times each spawn point was selected: `99995594, 99987497, 100000969, 99999733, 100012655, 100001681, 99999945, 99996286, 100004625, 100001015`. As you can see, numbers are very close to the mean value of 100000000 which means it is indeed uniformly random ;)

What you are probably seeing are the effects of "avoid closest" strategy which is by definition not uniformly random. Well, it *is* random among a subset of spawn points, but two spawn points on a map closest to other players are excluded from selection. Depending on where the players are, on a map with small total number of spawn points it is possible to skew the outcome to a certain subset of spawn points. That's way in 1v1 mode on maps with less than 6 spawn points there is an option to use uniformly random algorithm, while still using "avoid closest" strategy on larger maps. It's off by default, add 4 to `g_1v1_spawn_mode` value to activate that.

Also worth noting that original `SelectRandomDeathmatchSpawnPointAvoidingTwoClosest` algorithm contained a bug which incorrectly picked two closest spawn points. It is fixed in OpenTDM if `g_1v1_spawn_mode` value is not 0 or 4, or `g_tdm_spawn_mode` value is not 0.

Relevant code that shows how different options interact together is here (I don't know if it's documented somewhere):
https://github.com/skullernet/opentdm/blob/62d6d21bd315b54e42afc6947a1987ec6ad7a581/p_client.c#L938-L980

TL/DR: If you need truly random spawn point selection in TDM mode set `g_tdm_spawn_mode` to 2. But be prepared for telefrags, etc.
